Source/Codecov/Coverage/Tool/Coverage.cs

```diff
@@ -21,15 +22,45 @@ public Coverage(ICoverageOptions coverageOptions)
 
         private ICoverageOptions CoverageOptions { get; }
 
-        private static bool VerifyReportFile(string path)
+        private static bool VerifyReportFile(string path, out IEnumerable<string> expandedPath)
         {
+            HashSet<string> expanded = new HashSet<string> { path };
+            expandedPath = expanded;
             if (string.IsNullOrWhiteSpace(path))
             {
                 Logger.Log.Warning("Invalid report path.");
                 return false;
             }
 
-            if (!File.Exists(path))
+            if (path.Contains('*')
+                || path.Contains('?')
+                || path.Contains('!')
+                || path.Contains(','))
+            {
+                // Some basic guards
+                if (!path.EndsWith(".xml", StringComparison.OrdinalIgnoreCase))
```

Comment:
forcing a `.xml` extension on the paths is unfortunately a no go.
codecov don't just support xml reports, but also json, info and more.